HalClient.Net/HalHttpRequestException.cs

```diff
@@ -1,20 +1,27 @@
 using System;
 using System.Net;
+using System.Runtime.Serialization;
 using HalClient.Net.Parser;
 
 namespace HalClient.Net
 {
-	[Serializable]
-	public class HalHttpRequestException : Exception
-	{
-		public HalHttpRequestException(HttpStatusCode statusCode, string reason, IRootResourceObject resource = null)
-			: base($"{(int) statusCode} ({reason})")
-		{
-			StatusCode = statusCode;
-			Resource = resource;
-		}
+    [Serializable]
+    public class HalHttpRequestException : Exception
+    {
+#if REQUIRES_EXCEPTION_SERIALIZATION_CTOR
```

Comment:
@mookid8000 Why do we need a condition here? Would it harm to always have the constructor b/c of its argument types not being available in all frameworks?

Reply:
Yes, it's because the `StreamingContext` and `SerializationInfo` types are not available in .NET Standard 1.3, and it's not possible/necessary to create serialization constructors there.